Mimble Wimble.md

```diff
@@ -0,0 +1,124 @@
+
+## Could it work for Decred?
+
+Decred utilizes Stakeshuffle as its privacy protocol. It is responsible for mixing multiple transactions, and hiding who sent coins to whom, without the need for third parties. If you want to learn more about Stakeshuffle click here (https://docs.decred.org/privacy/cspp/overview/?ref=cypherpunktimes.com)!
+
+To understand more if Mimblewimble could be a good addition to the Decred blockchain I went through its implementation on Litecoin. Also highlighted some of its advantages and disadvantages.
+
+Mimbleimble was implemented on Litecoin by the use of a feature called Extension Blocks. The implementation launched in May 2022, and made it possible for Litecoin holders to use the new privacy features by moving their Litecoin into these Extension Blocks. These blocks are responsible for hiding the transaction information.
+
+To engage with this privacy feature, users perform a "pegging-in" transaction to move their Litecoin into an Extension Block, using a specific MimbleWimble Extension Block (MWEB) address. These transactions are facilitated by a special transaction known as the HogEx (Hogwarts Express), which processes both the pegging-in of funds to the Extension Block and the pegging-out back to the main, transparent blockchain. The HogEx also manages the Extension Block balance, adjusting it for incoming and outgoing transactions.
+
+While the majority of Litecoin transactions continue to operate on the transparent main blockchain, the introduction of MimbleWimble via Extension Blocks provides a significant step forward in offering privacy options to users.
+
+Implementing MimbleWimble on the Decred blockchain is theoretically possible. Decred governance model and adaptability could facilitate the consideration and potential implementation of MimbleWimble if there was enough community support.
+
+The technical integration of MimbleWimble would require a lot of development effort and a thorough review process to ensure that it aligns with Decred’s security, privacy, and scalability goals. One of the main considerations would be how to integrate MimbleWimble's privacy features with Decred's existing features, such as its governance system, and the Stakeshuffle protocol.
+
+There are also some Mimblewimble drawbacks that need to be taken into consideration: 
+
+Slower Transactions: Networks that use Confidential Transactions, like MimbleWimble, can be slower than others such as Ripple (XRP) and proof-of-stake networks like Solana and Cardano. This is because the data for Confidential Transactions is larger, which can slow things down.
+
+Quantum Computer Risk: MimbleWimble relies on digital signatures to work. This makes it potentially more at risk from future attacks by quantum computers, which might be able to break these digital signatures. Decred has overcome this problem with post-quantum secure mixing on stakeshuffle.
+
```

Comment:
Section "Could it work for Decred?"

I'm not an expert on MW but from what you've highlighted in this section, it seems like a global side chain, is this correct? a few questions:

* Do extension blocks remain open forever or is there some kind of closing mechanism?
* Does every user on the platform use the same extension blocks? e.g. is there just one set of extension blocks like a side chain or can users open their own extension blocks?
* Do full nodes running the network maintain the extension blocks?
* Do extension blocks have a size limit?

Reply:
Another consideration for this section. Is why would Decred need Mimble Wimble when it has the Lightning Network?

* The lightning network is very private
* Extremely scalable
* Cheap for transactions
* and has instant transactions